v2/types.go

```diff
@@ -1364,6 +1364,13 @@ type CandleSnapshot struct {
 	Snapshot []*Candle
 }
 
+type CandleHistoryParams struct {
```

Comment:
how do you divide parameters embedded in a CandleHistoryParams struct versus being on the function's call? it seems query parameters are in this struct, which fits the way REST works, but doesn't match other REST API patterns in this library (granted there don't seem to be many good examples besides the book call here: https://github.com/bitfinexcom/bitfinex-api-go/blob/master/v2/rest/book.go#L14).

it feels arbitrary to just put some parameters into a struct, I think putting these in a longer function would make it more consistent with the library.

Reply:
I was attempting to cater for the distinction between the mandatory (in the function call) and optional parameters (struct), although I see in some other people's implementations they just use a string map.

Happy to put them on the function though, have a preference for simpler, although is that going to make it impossible to make a call without supplying them?  Is that a use-case that matters?

Reply:
for history, I see the minimum set of params as just symbol and resolution. the API will give a valid response with only these valued supplied.  I ran your `rest-candles` example with an empty candle parameter struct, and the API returned 120 candles from 04/23/2018 @ 2:00 pm to 11:55 pm UTC.

You're right we would want to provide a fully functional long form call with all parameters. This is my idea at least, and it has been helpful with constructors in tests.  But if you can find supporting golang.org examples that show this isn't really idiomatic go, I'd be happy to take a second look at all of this.

Reply:
Thanks Matt, I'll adjust it tonight to go with the 2nd function call version, people can then either have minimum or maximum control.